packages/main/src/Tree.ts

```diff
@@ -319,6 +347,92 @@ class Tree extends UI5Element {
 		return !!this.header.length;
 	}
 
+	_ondragenter(e: DragEvent) {
+		e.preventDefault();
+	}
+
+	_ondragleave(e: DragEvent) {
+		if (e.relatedTarget instanceof Node && this.shadowRoot!.contains(e.relatedTarget)) {
+			return;
+		}
+
+		this.dropIndicatorDOM!.targetReference = null;
+	}
+
+	_ondragover(e: DragEvent) {
+		const draggedElement = DragRegistry.getDraggedElement();
+		const allItemsTraversed: Array<TreeItemBase> = []; // use the actual items for rearranging
+		const allLiNodesTraversed: Array<HTMLElement> = []; // use the only <li> nodes to determine positioning
+		if (!(e.target instanceof HTMLElement) || !draggedElement) {
+			return;
+		}
+
+		this.walk(item => {
+			allItemsTraversed.push(item);
+			allLiNodesTraversed.push(item.shadowRoot!.querySelector("li")!);
+		});
+
+		const closestPosition = findClosestPosition(
+			allLiNodesTraversed,
+			e.clientY,
+			Orientation.Vertical,
+		);
+
```

Comment:
Also check if there is attempt to drop node inside itself. In such cases, don't fire the `move-over` event